engine/src/main/java/org/pentaho/di/shared/DatabaseConnectionManager.java

```diff
@@ -68,12 +75,38 @@ private void populateDbMetaMap( Map<String, Node> nodesMap ) throws KettleXMLExc
 
   }
 
+  /**
+   * Add a new database connection. The new connection is added to xml file and also in the in memory map.
+   * @param databaseMeta
+   * @return boolean
+   * @throws KettleException
+   */
+  public synchronized boolean addDatabase( DatabaseMeta databaseMeta ) throws KettleException {
```

Comment:
yeah, these should have `@Override`, and probably have `@inheritdoc` in their javadoc.